### PR TITLE
CDPS-610: Implementing conditional reception move button

### PR DESCRIPTION
**Pull request by brightonsbox**
Provides the following link:

Screenshot 2024-01-26 at 12 00 45

See Jira for details of the conditions for the link.

### CODE REVIEW

File: assets/scss/application.scss [hunk 1]
```diff
@@ -35,18 +35,19 @@ $govuk-page-width: $moj-page-width;
...
-@import './pages/alerts-page';
+@import './pages/alerts';
 @import './pages/belief-history';
-@import './pages/case-notes-page';
+@import './pages/case-notes';
 @import './pages/money';
-@import './pages/overview-page';
-@import './pages/personal-page';
-@import './pages/work-and-skills-page';
-@import './pages/prisoner-schedule-page';
```
Comment (brightonsbox) [inline assets/scss/application.scss:49]: Just a renaming to make the '-page' or not '-page' consistent.

File: server/controllers/prisonerCellHistoryController.ts [hunk 2]
```diff
@@ -68,16 +68,15 @@ export default class PrisonerCellHistoryController {
...
       const name = formatName(firstName, middleNames, lastName, { style: NameFormatStyle.firstLast })
 
-      const prisonApiClient = this.prisonApiClientBuilder(clientToken)
+      const prisonApiClient = this.prisonApiClientBuilder(res.locals.clientToken)
 
-      const page = 0
-      const cells = await prisonApiClient.getOffenderCellHistory(bookingId, { page, size: 10000 })
+      const [cells, receptions] = await Promise.all([
+        prisonApiClient.getOffenderCellHistory(bookingId, { page: 0, size: 10000 }),
+        prisonApiClient.getReceptionsWithCapacity(prisonId),
```
Comment (brightonsbox) [inline server/controllers/prisonerCellHistoryController.ts:78]: As part of the follow up work to refactor this code, I'm going to look at conditionally calling the receptionsWithCapacity API, if the button isn't going to be displayed, we don't need the extra API call

File: server/controllers/prisonerCellHistoryController.test.ts [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+  canViewCellMoveButton: true,
+  canViewMoveToReceptionButton: true,
+  cellHistoryGroupedByAgency: [],
+  dpsBaseUrl: `${config.apis.dpsHomePageUrl}${profileUrl}`,
+  changeCellLink: `${config.apis.dpsHomePageUrl}${profileUrl}/cell-move/search-for-cell?returnUrl=${profileUrl}`,
+  moveToReceptionLink: `${config.apis.dpsHomePageUrl}${profileUrl}/reception-move/consider-risks-reception`,
+  currentLocation,
+  occupants: [],
+  prisonId: 'MDI',
+}
```
Comment (whitfield-mj) [inline server/controllers/prisonerCellHistoryController.test.ts:47]: The types of these mocks aren't particularly helpful, is there not a real type for them?

**Review by whitfield-mj — APPROVED** [review]

File: server/controllers/prisonerCellHistoryController.test.ts (hunk 3 above)
Comment (brightonsbox) [reply server/controllers/prisonerCellHistoryController.test.ts:47]: Agreed, I'm going to tackle this sort of thing in this ticket: https://dsdmoj.atlassian.net/browse/CDPS-373 which is going to be a refactor and implementation of tests for the rest of the page!

**Pull request merged**